server/mailet/integration-testing/src/test/java/org/apache/james/mailets/RemoteDeliveryErrorTest.java

```diff
@@ -250,7 +179,13 @@ public void remoteDeliveryShouldBounceWhenAlwaysRCPT421() throws Exception {
     @Test
     public void remoteDeliveryShouldBounceWhenAlwaysFROM421() throws Exception {
         with()
-            .body(ALWAYS_421_FROM_BEHAVIOR)
+            .body(MockSmtpBehaviors.builder()
```

Comment:
I will keep being annoying but the use of rest-assured makes things more painful than they could be.
By delivering a full API, we could get rid of the explicit builder-to-body conversion. We could also avoid repeating the endpoint.
Finally, we could check the server return status without repeating ourselves.

Reply:
@trantienduchn let's go for your Feign API. Can you take care of it? (next week?)

Reply:
Okay, That's fine for me!

Reply:
#2675